kernel/entries.ml

```diff
@@ -99,11 +99,14 @@ type primitive_entry = {
 type 'a proof_output = constr Univ.in_universe_context_set * 'a
 type 'a const_entry_body = 'a proof_output Future.computation
 
-type 'a constant_entry =
-  | DefinitionEntry of definition_entry
-  | OpaqueEntry of 'a const_entry_body opaque_entry
-  | ParameterEntry of parameter_entry
-  | PrimitiveEntry of primitive_entry
+(** Dummy wrapper type discriminable from unit *)
+type 'a wrap = { wrap : 'a }
```

Comment:
Wouldn't it make sense to call this box "effects" ? Looks like unit means pure after all, while this thing is for impure stuff.
Or maybe just use an inductive with two constructors Pure != Effectful 'a.

(if it deserves a comment it means it is surprising)

Reply:
This is actually temporary, I have a further path that removes this type. I had to introduce it not to have to change too much the upper layers.

Reply:
I would still prefer that we give it a more meaningful name. Temporary things tend to stay longer than we expect...